doc/api/cli.md

```diff
@@ -1370,8 +1370,15 @@ added: v12.0.0
 -->
 
 This configures Node.js to interpret `--eval` or `STDIN` input as CommonJS or
-as an ES module. Valid values are `"commonjs"` or `"module"`. The default is
-`"commonjs"`.
+as an ES module. Valid values are `"commonjs"`, `"module"`, `"module-typescript"` and `"commonjs-typescript"`.
+The `"-typescript"` values are available only in combination with the flag `--experimental-strip-types`.
+The default is `"commonjs"`.
+
+If `--experimental-strip-types` is enabled and `--input-type`is not provided,
+Node.js will first try to run the code as JavaScript,
+then it will try to run the code as TypeScript.
+When using TypeScript, the `--input-type=type` flag should always be used,
+to reduce the overhead of multiple syntax detections.
```

Comment:
This is a bit strong. I don’t think people using `--eval` are generally concerned about performance, so parsing a <1 KB string multiple times isn’t a big deal. Rather than making a strong recommendation, we can just offer a hint on how to avoid the performance hit for those who care:

```suggestion
To avoid the delay of multiple syntax detection passes, the `--input-type=type` flag can be used to specify
how the `--eval` input should be interpreted.
```